examples/arduino-uno/src/bin/uno-ext-interrupt.rs

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Blinks a 4 leds in sequence on pins D3 - D6. When an external interrupt on D2/INT0 comes in
+ * the sequence is reversed.
+ * 
+ * Note: The use of the either crate requires the deactivation of std to use it in core. See the Cargo.toml 
+ * in this directory for details.
+ */
+#![no_std]
+#![no_main]
+#![feature(abi_avr_interrupt)]
+
+use panic_halt as _;
+use core::ops::Range;
+use core::sync::atomic::{AtomicBool, Ordering};
+use arduino_hal::port::{mode, Pin};
+use either::*;
+
+static REVERSED: AtomicBool = AtomicBool::new(false);
+
+fn is_reversed() -> bool {
+    return avr_device::interrupt::free(|_| { REVERSED.load(Ordering::SeqCst) });
```

Comment:
This critical section is superfluous - the "atomic" is synchronized on its own.

("atomic" really means "emulated atomic" because AVR does not have real atomic operations - instead, there are compiler intrinsics which emulate atomic operations by turning off interrupts)

Reply:
Fixed